metadata-ingestion/src/datahub/ingestion/source/usage/usage_common.py

```diff
@@ -45,19 +50,35 @@ def add_read_entry(
         for column in fields:
             self.columnFreq[column] += 1
 
+    def trim_query(self, query: str, budget_per_query: int) -> str:
+        trimmed_query = query
+        if len(query) > budget_per_query:
+            if budget_per_query - self.query_trimmer_string_space > 0:
+                end_index = budget_per_query - self.query_trimmer_string_space
+                trimmed_query = query[:end_index] + self.query_trimmer_string
+            else:
+                raise Exception(
+                    "Budget per query is too low. Please, decrease the number of top_n_queries."
+                )
```

Comment:
in what cases would this exception get raised? I would think this would never be a problem unless `n` was greater than the number of characters allowed in the budget?

Reply:
Normally this should not get raised as the config validation should catch this. 
This makes sure the budget per query is not too small and does not end up with a negative value which can cause to copy the whole string as `query[:-1]` means everything except the last character.